hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,570 @@
+        list_not_title = ['lang', 'company_type', 'email', 'website']
+
+        for key, value in contact_data.items():
+            if isinstance(value, str) and key not in list_not_title:
+                    contact_data[key] = value.title()
+
+        contact_id = request.env['res.partner'].create(contact_data)
+
+        # Utilizar logger en lugar de print
+        logger.info("Nuevo contacto creado con ID %s", contact_id.id)
+        
+        return {"message": f"Nuevo contacto creado con ID: {contact_id.id}", "contact_id": contact_id.id}
+    
+    @route('/update_contact', methods=['POST'], type='json', auth='user')
+    def update_contact(self):
+
+        data = request.jsonrequest
+        partner_id = data.get('partner_id')  # ID del contacto a actualizar
+        update_vals = data.get('update_vals')  # Valores a actualizar
+
+        # Encontrar el contacto por su ID
+        partner = request.env['res.partner'].browse(partner_id)
+
+        list_not_title = ['lang', 'company_type', 'email', 'website', 'vat']
+
+        for key, value in update_vals.items():
+            if isinstance(value, str) and key not in list_not_title:
+                    update_vals[key] = value.title()
+
+        # Actualizar el contacto con los nuevos valores
+        partner.write(update_vals)
+
+        logger.info("Contacto actualizado con ID %s", partner_id)
+
+        return {"message": f"Contacto con ID: {partner_id} actualizado exitosamente."}
+    
+    @route('/manage_contact', methods=['POST'], type='json', auth='user')
+    def manage_contact(self):
+        
+        data = request.jsonrequest
+        email = data.get('email')
+        phone = data.get('phone')
+        contact_data = data.get('contact_data', {})  # Datos del contacto a actualizar/crear
+
+        # Verificar si ya existe un contacto a partir de los primeros digitos del correo y ultimos digitos del telefono
+        email_prefix = email[0:5]
+        phone_suffix = phone[len(phone)-5:len(phone)]
+
+        # Crear dominios de búsqueda usando patrones con comodines
+        domain = [('email', 'ilike', f'{email_prefix}%'), ('phone', 'ilike', f'%{phone_suffix}')]
+
+        # Realizar la búsqueda con el dominio definido
+        existing_contact = request.env['res.partner'].search(domain, limit=1)
+
+
+        list_not_title = ['lang', 'company_type', 'email', 'website', 'phone']
+
+        for key, value in contact_data.items():
+            if isinstance(value, str) and key not in list_not_title:
+                    contact_data[key] = value.title()  # Convierte la primera letra de cada palabra a mayúscula
+
+        if existing_contact:
+            # Actualizar el contacto existente si se encontró
+            existing_contact.write(contact_data)
+            logger.info("Contacto actualizado con ID %s", existing_contact.id)
+            return {"message": f"Contacto actualizado con ID: {existing_contact.id}.", "contact_id": existing_contact.id}
+        else:
```

Comment:
```suggestion
```
 Unnecessary "else" after "return", remove the "else" and de-indent the code inside it